unittest/tsid-formulation.cpp

```diff
@@ -475,6 +477,165 @@ BOOST_AUTO_TEST_CASE ( test_invdyn_formulation_acc_force )
+  double w_posture = 1e-3;                // weight of joint posture task
+  double w_forceRef = 1e-5;               // weight of force regularization task
+  double w_RF = 1.0;                      // weight of right foot motion task
+  double kp_contact = 100.0;              // proportional gain of contact constraint
+  double kp_com = 1.0;                    // proportional gain of center of mass task
+
+  vector<string> package_dirs;
+  package_dirs.push_back(quadruped_model_path);
+  string urdfFileName = package_dirs[0] + "/urdf/quadruped.urdf";
+  RobotWrapper robot(urdfFileName,
+                     package_dirs,
+                     se3::JointModelFreeFlyer(),
+                     false);
+
+  BOOST_REQUIRE(robot.model().existFrame(frameName));
+
+  Vector q = robot.model().neutralConfiguration;
+  Vector v = Vector::Zero(robot.nv());
+  const unsigned int nv = robot.nv();
+
+  // Create initial posture.
+  q(0) = 0.1;
+  q(2) = 0.5;
+  q(6) = 1.;
+  for (int i = 0; i < 4; i++) {
+    q(7 + 2 * i) = -0.4;
+    q(8 + 2 * i) = 0.8;
+  }
+
+  // Create the inverse-dynamics formulation
+  InverseDynamicsFormulationAccForce *tsid =
+      new InverseDynamicsFormulationAccForce("tsid", robot);
+  tsid->computeProblemData(t, q, v);
+  const se3::Data & data = tsid->data();
+
+  // Place the robot onto the ground.
+  se3::SE3 fl_contact = robot.position(data, robot.model().getJointId("FL_contact"));
+  q[2] -= fl_contact.translation()(2);
+  tsid->computeProblemData(t, q, v);
+
+  // Add task for the COM.
+  TaskComEquality *comTask = new TaskComEquality("task-com", robot);
+  comTask->Kp(kp_com * Vector::Ones(3));
+  comTask->Kd(2.0 * comTask->Kp().cwiseSqrt());
+  tsid->addMotionTask(*comTask, w_com, 1);
+
+  Vector3 com_ref = robot.com(data);
+  TrajectoryBase *trajCom = new TrajectoryEuclidianConstant("traj_com", com_ref);
+  TrajectorySample sampleCom(3);
+  sampleCom = trajCom->computeNext();
+  comTask->setReference(sampleCom);
+
+  // Add contact constraints.
+  std::string contactFrames[] = {
+    "BL_contact", "BR_contact", "FL_contact", "FR_contact"
+  };
+
+  Vector3 contactNormal = Vector3::UnitZ();
+  std::array<ContactPoint*, 4> contacts;
+
+  for (int i = 0; i < 4; i++) {
+    ContactPoint* cp = new ContactPoint("contact_" + contactFrames[i], robot,
+        contactFrames[i], contactNormal, 0.6, 0., 10., w_forceRef);
+    cp->Kp(kp_contact*Vector::Ones(6));
+    cp->Kd(2.0*cp->Kp().cwiseSqrt());
+    cp->setReference(robot.position(data, robot.model().getJointId(contactFrames[i])));
+    cp->useLocalFrame(false);
+    tsid->addRigidContact(*cp, 1);
+
+    contacts[i] = cp;
+  }
+
+  // Create an HQP solver
+  SolverHQPBase * solver = SolverHQPFactory::createNewSolver(SOLVER_HQP_EIQUADPROG,
+                                                               "solver-eiquadprog");
+  solver->resize(tsid->nVar(), tsid->nEq(), tsid->nIn());
+
+  Vector dv = Vector::Zero(nv);
+  const HQPOutput *sol;
+  for (int i = 0; i < 10; i++) {
```

Comment:
For coherence with the other tests you should use `max_it` here instead of the hard-coded value 10